### PR TITLE
OKX: Change float64 to convert.StringtoFloat64 in okx_types

### PR DESCRIPTION
**Pull request by Beadko**
# PR Description

Change type float64 to convert.StringToFloat64

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/okx/okx_types.go [hunk 1]
```diff
@@ -314,9 +314,9 @@ type Instrument struct {
 
 // DeliveryHistoryDetail holds instrument id and delivery price information detail
 type DeliveryHistoryDetail struct {
-	Type          string  `json:"type"`
-	InstrumentID  string  `json:"insId"`
-	DeliveryPrice float64 `json:"px,string"`
+	Type          string                  `json:"type"`
+	InstrumentID  string                  `json:"insId"`
```
Comment (shazbert) [inline exchanges/okx/okx_types.go:318]: Add this line to exclude_lines.txt so this linter can be skipped. The includes the tab space as well.

File: exchanges/okx/okx_types.go [hunk 2]
```diff
@@ -3039,14 +3039,14 @@ type SubAccountAPIKeyResponse struct {
 type MarginBalanceParam struct {
 	AlgoID     string  `json:"algoId"`
 	Type       string  `json:"type"`
-	Amount     float64 `json:"amt,string"`               // Adjust margin balance amount Either amt or percent is required.
-	Percentage float64 `json:"percent,string,omitempty"` // Adjust margin balance percentage, used In Adjusting margin balance
+	Amount     float64 `json:"amt,string"`     // Adjust margin balance amount Either amt or percent is required.
+	Percentage float64 `json:"percent,string"` // Adjust margin balance percentage, used In Adjusting margin balance
```
Comment (shazbert) [inline exchanges/okx/okx_types.go:3043]: just checking Percentage is a requirement?

**Review by shazbert — CHANGES_REQUESTED** [review]
Good stuff Bea, just one thing.

File: exchanges/okx/okx_types.go (hunk 1 above)
Comment (Beadko) [reply exchanges/okx/okx_types.go:318]: done

File: exchanges/okx/okx_types.go (hunk 2 above)
Comment (Beadko) [reply exchanges/okx/okx_types.go]: that was a typo, sorry. I have fixed it

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for doing that @Beadko

**Review by thrasher- — CHANGES_REQUESTED** [review]
@Beadko thanks for this PR. There's a merge conflict that needs to be resolved and then after that it should be good to go

**Comment by Beadko** [comment]
Rebased on master. Expecting to rebase again when #1415 is merged.

**Review by gloriousCode — APPROVED** [review]
retACK! Thanks for the rebase!

**Review by thrasher- — APPROVED** [review]
Thanks @Beadko !

**Pull request merged**